src/NumericalAlgorithms/Interpolation/LinearInterpolationND.hpp

```diff
@@ -0,0 +1,374 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <memory>
+
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Requires.hpp"
+
+namespace intrp {
+
+template <size_t Dimension, size_t NumberOfVariables, bool UniformSpacing>
+class LinearSpanInterpolationND {
+ public:
```

Comment:
We prefer
```cpp
class Blah {
 public:
  /* all public functions and members */

 private:
  /* all private members */
};
```
unless there's a techincal reason why the ordering can't be that way.